paimon-core/src/main/java/org/apache/paimon/operation/KeyValueFileStoreScan.java

```diff
@@ -121,10 +134,33 @@ protected boolean filterByStats(ManifestEntry entry) {
         }
 
         return filter.test(
-                entry.file().rowCount(),
-                serializer.evolution(stats.minValues()),
-                serializer.evolution(stats.maxValues()),
-                serializer.evolution(stats.nullCounts(), entry.file().rowCount()));
+                        entry.file().rowCount(),
+                        serializer.evolution(stats.minValues()),
+                        serializer.evolution(stats.maxValues()),
+                        serializer.evolution(stats.nullCounts(), entry.file().rowCount()))
+                && (!fileIndexReadEnabled || testFileIndex(entry.file().embeddedIndex(), entry));
+    }
+
+    private boolean testFileIndex(@Nullable byte[] embeddedIndexBytes, ManifestEntry entry) {
+        if (embeddedIndexBytes == null) {
+            return true;
```

Comment:
If fileIndex does not exist in embeddedIndex but in a separate file, what logic of filter by FileIndex in pk table with DV?